url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgReport.java

```diff
@@ -0,0 +1,201 @@
+/**
+ *+
+ *	ArgReport.java
+ *	1.0.0  2014-08-02  Leo Hinterlang
+ *-
+ */
+package com.fidelis.argface;
+
+import java.util.List;
+
+/**
+ * Report on ArgFace results.
+ * 
+ * @version 1.0.0
+ * @author Leo Hinterlang
+ */
+public class ArgReport {
+	
+	/**
+	 * Reports on the results of command line parsing.
+	 * The selection indicates the desired output. These include:
+	 * <ul>
+	 * <li>all - all information.</li>
+	 * <li>has - only items that appear on the command line.</li>
+	 * <li>options - just the options</li>
+	 * <li>operands - just the operands</li>
+	 * <li>has-options - options that appear on the command line.</li>
+	 * <li>has-operands - operands that appear on the command line.</li>
+	 * <li>-option - a named option.</li>
+	 * <li>&lt;operand&gt; - a named operand.</li>
+	 * </ul>
+	 *
+	 * @param selection indicates the desired output
+	 * @param base the ArgBase class
+	 */
+	static void report (String selection, ArgBase base) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.